slowapi/extension.py

```diff
@@ -493,8 +514,13 @@ def __check_request_limit(
                     else []
                 )
                 all_limits += route_limits
-                if not route_limits and not (
-                    in_middleware and name in self.__marked_for_limiting
+                combined_defaults = all(
+                    not limit.override_defaults for limit in route_limits
+                )
+                if (
+                    not route_limits
+                    and not (in_middleware and name in self.__marked_for_limiting)
+                    or combined_defaults
                 ):
                     all_limits += list(itertools.chain(*self._default_limits))
```

Comment:
does less than the original https://github.com/alisaifee/flask-limiter/blob/55df08f14143a7e918fc033067a494248ab6b0c5/flask_limiter/extension.py#L592 but i _think_ it's sufficient for currently supported functionality.

Reply:
I can find a mistake in the logic either (It doesn't mean there isn't one :wink: ). I think it would be worth spending some time commenting this bit of code, as I find it pretty obscure, with all these `xxxx_limits` variables. And we should add some more tests, but I'm happy to merge this at this point, as the middleware is pretty cool functionality.
Is that ok with you?

Reply:
sounds good to me :)

Reply:
i think one useful thing to do might be to implement some more of the flask-limiter tests with `@pytest.mark.xfail` so it's more obvious how much of the api is implemented and how far from feature parity we are

Reply:
I'll have a go at that when I find some time!

Reply:
added https://github.com/laurentS/slowapi/issues/17 off the back of this